tools/c7n_gcp/tests/test_gke.py

```diff
@@ -67,16 +72,70 @@ def test_cluster_get(self):
             ],
         )
 
+    def test_cluster_set_labels(self):
+        project_id = 'cloud-custodian'
+        name = "standard-cluster-1"
+        factory = self.replay_flight_data('gke-cluster-set-label', project_id)
+        p = self.load_policy(
+            {
+                'name': 'label-gke-cluster',
+                'resource': 'gcp.gke-cluster',
+                'filters': [{'name': name}],
+                'actions': [{'type': 'set-labels',
+                            'labels': {'test_label': 'test_value'}}]},
+            session_factory=factory
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
This looks good, though I think if we want to work the fingerprint mismatch logic in this test we need to:

- Enable caching (tests default to a null cache configuration)
- Run 2 policies with the `set-labels` action

For example:

```suggestion
        p = self.load_policy(
            {
                'name': 'label-gke-cluster',
                'resource': 'gcp.gke-cluster',
                'filters': [{'name': name}],
                'actions': [{'type': 'set-labels',
                            'labels': {'test_label': 'test_value'}}]},
            cache=True,
            config=Config.empty(
                cache='memory',
                cache_period=10,
                output_dir=self.get_temp_dir(),
            ),
            session_factory=factory
        )
        resources = p.run()
        self.assertEqual(len(resources), 1)
        p = self.load_policy(
            {
                'name': 'relabel-gke-cluster',
                'resource': 'gcp.gke-cluster',
                'filters': [{'name': name}],
                'actions': [{'type': 'set-labels',
                            'labels': {'test_label': 'new_value'}}]},
            cache=True,
            config=Config.empty(
                cache='memory',
                cache_period=10,
                output_dir=self.get_temp_dir(),
            ),
            session_factory=factory
        )
        resources = p.run()
        self.assertEqual(len(resources), 1)
```